[TensorExpr] Don't rely on exceptions in Vectorizer.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64609

We've been using exceptions to indicate whether vectorization succeeded
or not, but that posed some problems with (e.g. we spent too much time
symbolicazing these exceptions). This change converts this mechanism to
a standard error return code.

Differential Revision: [D30795342](https://our.internmc.facebook.com/intern/diff/D30795342)

Commit:
[TensorExpr] Don't rely on exceptions in Vectorizer.

We've been using exceptions to indicate whether vectorization succeeded
or not, but that posed some problems with (e.g. we spent too much time
symbolicazing these exceptions). This change converts this mechanism to
a standard error return code.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30795342).

Commit:
Update on "[TensorExpr] Don't rely on exceptions in Vectorizer."


We've been using exceptions to indicate whether vectorization succeeded
or not, but that posed some problems with (e.g. we spent too much time
symbolicazing these exceptions). This change converts this mechanism to
a standard error return code.

Differential Revision: [D30795342](https://our.internmc.facebook.com/intern/diff/D30795342)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30795342).

Commit:
Update on "[TensorExpr] Don't rely on exceptions in Vectorizer."


We've been using exceptions to indicate whether vectorization succeeded
or not, but that posed some problems with (e.g. we spent too much time
symbolicazing these exceptions). This change converts this mechanism to
a standard error return code.

Differential Revision: [D30795342](https://our.internmc.facebook.com/intern/diff/D30795342)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30795342).

bertmaher reviewed (approved):
Cool, this is way less invasive a change than I would have expected!

Commit:
clang-format on "[TensorExpr] Don't rely on exceptions in Vectorizer."


We've been using exceptions to indicate whether vectorization succeeded
or not, but that posed some problems with (e.g. we spent too much time
symbolicazing these exceptions). This change converts this mechanism to
a standard error return code.

Differential Revision: [D30795342](https://our.internmc.facebook.com/intern/diff/D30795342)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30795342).

Commit:
Update on "[TensorExpr] Don't rely on exceptions in Vectorizer."


We've been using exceptions to indicate whether vectorization succeeded
or not, but that posed some problems with (e.g. we spent too much time
symbolicazing these exceptions). This change converts this mechanism to
a standard error return code.

Differential Revision: [D30795342](https://our.internmc.facebook.com/intern/diff/D30795342)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30795342).
